vector/src/main/java/im/vector/app/features/home/room/detail/RoomDetailViewModel.kt

```diff
@@ -196,6 +199,24 @@ class RoomDetailViewModel @AssistedInject constructor(
         if (OutboundSessionKeySharingStrategy.WhenEnteringRoom == BuildConfig.outboundSessionKeySharingStrategy && room.isEncrypted()) {
             prepareForEncryption()
         }
+
+        if (initialState.switchToParentSpace) {
+            // We are coming from a notification, try to switch to the most relevant space
+            // so that when hitting back the room will appear in the list
+            appStateHandler.getCurrentRoomGroupingMethod()?.space().let { currentSpace ->
+                val currentRoomSummary = room.roomSummary() ?: return@let
+                // nothing we are good
+                if (currentSpace == null || !currentRoomSummary.flattenParentIds.contains(currentSpace.roomId)) {
+                    // take first one or switch to home
+                    appStateHandler.setCurrentSpace(
+                            currentRoomSummary
+                                    .flattenParentIds.firstOrNull { it.isNotBlank() },
+                            // force persist, because if not on resume the AppStateHandler will resume
+                            // the current space from what was persisted on enter background
+                            persistNow = true)
```

Comment:
It looks like a weird workaround.
Do we want to update the notifications to group them by space instead, and/or include a spaceId in the notification intent itself?